llama_index/embeddings/huggingface.py

```diff
@@ -201,3 +212,65 @@ def _get_text_embeddings(self, texts: List[str]) -> List[List[float]]:
         """Get text embeddings."""
         texts = [self._format_text(text) for text in texts]
         return self._embed(texts)
+
+
+class HuggingFaceInferenceAPIEmbeddings(HuggingFaceInferenceAPI, Embeddings):
+    """
+    Wrapper on the Hugging Face's Inference API for embeddings.
+
+    Overview of the design:
+    - Implemented against LangChain's embeddings
+    - Uses the feature extraction task: https://huggingface.co/tasks/feature-extraction
+    """
+
+    pooling: "Optional[Pooling]" = Field(
+        default=None,
```

Comment:
Does it make more sense to have this be a default factory for `CLS` ? It feels weird to leave pooling `None` 🤔

Reply:
Yeah my thoughts with `None` were it's the least presumptive default, it will ensure people know what's going on.

The default works fine for some models that don't output a multi-dimensional embedding.

I can change it to `CLS` if you think it's the best idea, not opinionated here

Reply:
Let's change it to CLS. In the future, if we need to, we can provide more guidance on this, but most users aren't going to know the difference, and most embeddings models work best with CLS pooling

Reply:
Alright, moved it to CLS just now